lib/validation/core/validator.ts

```diff
@@ -4,31 +4,31 @@ import { Validated } from './validated';
 
 export type ValidationPredicate<T> = (t: T) => boolean;
 
-export class Validation<T> {
+export class Validation {
   tag: string;
-  target: T;
-  predicate: ValidationPredicate<T>;
+  target: unknown;
+  predicate: ValidationPredicate<unknown>;
   message: string;
```

Comment:
The unknowns above can have a type since we know the types allowed for validation.